boot.js

```diff
@@ -3112,11 +1429,14 @@ function configurePackage(location, description, parent) {
     // Deal with redirects
     var redirects = description.redirects;
     if (redirects !== void 0) {
-        Object.keys(redirects).forEach(function (name) {
-            modules[name] = {
-                id: name,
-                redirect: Identifier.normalize(Identifier.resolve(redirects[name], "")),
-                location: URL.resolve(location, name)
+        Object.keys(redirects).forEach(function (source) {
+            var target = redirects[source];
+            source = Identifier.normalize(Identifier.resolve(source, ""));
+            target = Identifier.normalize(Identifier.resolve(target, ""));
+            modules[source] = {
+                id: source,
+                redirect: target,
+                location: URL.resolve(location, target)
```

Comment:
This fixes ASAP now that ASAP uses redirects that are compatible with Browserify. The keys of browserify redirects are relative for local and top-level for for other packages or Node.js internals. Mr has a single namespace for both.